ocaml/xapi-idl/lib/task_server.ml

```diff
@@ -126,12 +127,22 @@ functor
 
     (* [add dbg f] creates a fresh [t], registers and returns it *)
     let add tasks dbg (f : task_handle -> Interface.Task.async_result option) =
+      let dbg, tracing =
+        match String.index_opt dbg '\n' with
```

Comment:
If this is splitting some string, using Astring functions would be more obvious. Another option is to use sscanf.

Reply:
split_on_char could also be used here and pattern match.

Reply:
Hmm... I put the tracing data in the `dbg` string for the hackathon project as a quick hack, which we would consider replacing with proper types when "productising". Either by adding a new function parameter for the tracing data to every function, or changing the type of `dbg` to a tuple or record. This may or may not be easy :) but worth thinking about.